### PR TITLE
Add  DALL·E 3 functionality

### PR DESCRIPTION
**Pull request by Gary-H9**
Hello @alexrudall. I've created this PR to _make a start_ on making the DALLE 3 available in the gem. 

I think this PR has some way to go. In particular, I'm not sure how/if changes should be made to `lib/images.rb`. 

I've made an addition to the spec for images. In particular I've broken out the functionality of both DALLE2 and DALLE3. At the moment there is only the ability to generate images in DALLE3 hence the omission of edits and variations. 

I added a new tests for DALL·E 3 and my VCR recording. 

I've also made a start on updating the README.md. 

If you could let me know if this is appropriate or what else might need done I will be happy to take this as far as I can.

## All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](../blob/main/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you added an explanation of what your changes do and why you'd like us to include them?

### CODE REVIEW

**Comment by alexrudall** [comment]
Thanks! This looks good to me, although the cassettes aren't working on CI for some reason. 🦁

**Comment by Gary-H9** [comment]
👋🏻 Yeah I'm not overly sure why that's happened. I'll have another look over this weekend. 

Does anything else need edited anywhere? I'm unsure about what 'lib/images.rb.' is doing in all honesty... 😂

**Comment by alexrudall** [comment]
- You on VCR 1? Might need to be on VCR 2 or something
- Can you edit/make variations on DALLE 3? If not, that should probably go on the docs. Or maybe just raise an error in those methods in `images.rb`

File: README.md [hunk 1]
```diff
@@ -705,8 +705,11 @@ Note that you have 10 minutes to submit your tool output before the run expires.
 
 ### Image Generation
 
-Generate an image using DALL·E! The size of any generated images must be one of `256x256`, `512x512` or `1024x1024` -
-if not specified the image will default to `1024x1024`.
+Generate images using DALL·E 2 or DALL·E 3! 
+
+#### DALL·E 2
```
Comment (alexrudall) [inline README.md:710]: Could ya put these headings in the README index please, at the top?

Comment (Gary-H9) [reply README.md:712]: Sure thing, [added](https://github.com/alexrudall/ruby-openai/pull/418/commits/0e86222970e0e077d913d8922cc2663106dcb554)them in now.

File: README.md [hunk 2]
```diff
@@ -38,6 +38,8 @@ Stream text with GPT-4, transcribe and translate audio with Whisper, or create i
 - [Runs](#runs)
   - [Runs involving function tools](#runs-involving-function-tools)
 - [Image Generation](#image-generation)
+  - [DALL·E 2](#dalle-2)
+  - [DALL·E 2](#dalle-3)
```
Comment (alexrudall) [inline README.md:42]:
```suggestion
  - [DALL·E 3](#dalle-3)
```

File: README.md (hunk 1 above)
Comment (alexrudall) [reply README.md:712]: ta! small typo

**Comment by Gary-H9** [comment]
👋  Hello @alexrudall, sorry for the delay looking at this. I've updated the cassettes. (No real idea what was going on on the first occasion. Deletion and re-running seems to have solved the CircleCI issues.)

I've also updated the tests / prompt to be a springer spaniel, in tribute to mine. 🫶🐶

If anyone else needs this in future I created this quick script to generate the image: 
```ruby
require "openai"

client = OpenAI::Client.new(access_token: "<my-api-token>")

response = client.images.generate(parameters: {prompt: "A springer spaniel cooking pasta wearing a hat of some sort", size: "1024x1792", model: "dall-e-3", quality: "hd"})
puts response.dig("data", 0, "url")
```

**Comment by jtoy** [comment]
should we mention in README for now that its not supported?

**Comment by Gary-H9** [comment]
Unsure what you are referencing @jtoy. 

Anything else needed on this @alexrudall?

**Comment by jtoy** [comment]
@Gary-H9 I mean that we don't currently have dalle3 support yet.

**Comment by jtoy** [comment]
@Gary-H9 fyi, i deployed your fork to production and it works great!

**Comment by dcrec1** [comment]
Can this be merged?

**Comment by alexrudall** [comment]
Released in [v6.4](https://github.com/alexrudall/ruby-openai/releases/tag/v6.4.0)

**Pull request merged**